plotly/tools.py

```diff
@@ -1230,3 +1230,35 @@ def __init__(self, url, width, height):
 
         def _repr_html_(self):
             return self.embed_code
+
+
+def return_figure_from_figure_or_data(figure_or_data, validate_figure):
+    if isinstance(figure_or_data, dict):
+        figure = figure_or_data
+    elif isinstance(figure_or_data, list):
+        figure = {'data': figure_or_data}
+    else:
+        raise exceptions.PlotlyError("The `figure_or_data` positional "
+                                     "argument must be either "
+                                     "`dict`-like or `list`-like.")
+    if validate_figure:
+        try:
+            validate(figure, obj_type='Figure')
```

Comment:
that darn `tools.validate`. if that didn't exist, i woulda placed this function in `utils.py` since i can't imagine anyone actually needing it.